core/txpool/subpool.go

```diff
@@ -71,6 +71,10 @@ type LazyResolver interface {
 // may request (and relinquish) exclusive access to certain addresses.
 type AddressReserver func(addr common.Address, reserve bool) error
 
+// IsAddressReserved is passed by the main transaction pool to subpools, so they
+// can query whether the address has been reserved or not.
+type IsAddressReserved func(addr common.Address) bool
```

Comment:
Thoughts on rolling this and the above `AddressReserver` into a struct that holds the map and lock and expose some methods `Reserve` and `Reserved`? Now that their are two functions operating on top of the map, it seems simpler in the long run to consolidate them into an first call object.

I can take a stab at this if it seems reasonable.

Reply:
yep, sounds reasonable to me!